### PR TITLE
feat(system_monitor): add on/off config for network traffic monitor

### PR DESCRIPTION
**Pull request by iwatake2222**
### Why this PR is needed

- System Monitor (Net Monitor) uses `nethogs` to monitor network traffic by program process
- However, the CPU usage of `nethogs` is high in some systems
- Monitoring network traffic by program is not mandatory, while network usage/errors are useful metrics
- Therefore, a config to on/off `nethogs` is required

### What this PR changes

- Add a new config named `enable_traffic_monitor` into `net_monitor.param.yaml`
- Stop a thread calling `nethogs` in `traffic_reader` service when `enable_traffic_monitor` is `false`
- Fix a termination logic for `nethogs` command
  - Originally, `c.terminate()` is used to stop `nethogs` command. However, its parent process (`traffic_reader`) is a daemon process and such process won't receive SIGCHLD. As a result, `nethogs` command becomes a zombie process
  - SIGKILL is sent to stop `nethogs` command, instead of `c.terminate()`
  - The following screen shot shows `nethogs` becomes zombie process after restarting Autoware
  - image

## Related links

None

## How was this PR tested?

- `nethogs` command doesn't run when `enable_traffic_monitor` is `false` 

image

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by iwatake2222** [comment]
@ito-san @TetsuKawa 
Could you review this PR?

File: system/system_monitor/src/net_monitor/net_monitor.cpp [hunk 1]
```diff
@@ -81,7 +82,11 @@ NetMonitor::NetMonitor(const rclcpp::NodeOptions & options)
   get_reassembles_failed_column_index();
 
   // Send request to start nethogs
-  send_start_nethogs_request();
+  if (enable_traffic_monitor_) {
+    send_start_nethogs_request();
+  } else {
+    send_stop_nethogs_request();
```
Comment (TetsuKawa) [inline system/system_monitor/src/net_monitor/net_monitor.cpp:88]: The current function name feels off. I think send_skip_nethogs_request would be better.

File: system/system_monitor/include/system_monitor/traffic_reader/traffic_reader_service.hpp [hunk 2]
```diff
@@ -69,6 +69,11 @@ class TrafficReaderService
    */
   void start_nethogs(boost::archive::text_iarchive & archive);
 
+  /**
+   * @brief Stop nethogs
+   */
+  void stop_nethogs();
```
Comment (TetsuKawa) [inline system/system_monitor/include/system_monitor/traffic_reader/traffic_reader_service.hpp:75]: A Same point.`skip_nethogs`

File: system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp [hunk 3]
```diff
@@ -120,6 +121,9 @@ void TrafficReaderService::handle_message(const char * buffer)
     case Request::GET_RESULT:
       get_result();
       break;
+    case Request::STOP_NETHOGS:
+      stop_nethogs();
```
Comment (TetsuKawa) [inline system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp:125]: A Same point. `skip_nethogs`, `SKIP_NETHOGS`

File: system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp [hunk 4]
```diff
@@ -153,6 +157,19 @@ void TrafficReaderService::start_nethogs(boost::archive::text_iarchive & archive
   thread_ = std::thread(&TrafficReaderService::execute_nethogs, this);
 }
 
+void TrafficReaderService::stop_nethogs()
+{
+  syslog(LOG_INFO, "Stopping nethogs...\n");
```
Comment (TetsuKawa) [inline system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp:162]: A Same Point. `Skipping`

File: system/system_monitor/src/net_monitor/net_monitor.cpp [hunk 5]
```diff
@@ -624,6 +634,21 @@ void NetMonitor::send_start_nethogs_request()
...
+void NetMonitor::send_stop_nethogs_request()
+{
+  // Connect to boot/shutdown service
+  if (!connect_service()) {
+    close_connection();
+    return;
+  }
+
+  // Send data to traffic-reader service
+  send_data(traffic_reader_service::STOP_NETHOGS);
```
Comment (TetsuKawa) [inline system/system_monitor/src/net_monitor/net_monitor.cpp:646]: A Same Point. `SKIP_NETHOGS`

**Review by TetsuKawa — COMMENTED** [review]
There were no issues in terms of functionality. However, the term `stop_nethogs` feels a bit off. How about changing it to `skip_nethogs`?

File: system/system_monitor/src/net_monitor/net_monitor.cpp (hunk 1 above)
Comment (iwatake2222) [reply system/system_monitor/src/net_monitor/net_monitor.cpp]: Fixed in fad266d995c46e12f8ff763e4db67cb19a64298d

File: system/system_monitor/include/system_monitor/traffic_reader/traffic_reader_service.hpp (hunk 2 above)
Comment (iwatake2222) [reply system/system_monitor/include/system_monitor/traffic_reader/traffic_reader_service.hpp]: Fixed in fad266d995c46e12f8ff763e4db67cb19a64298d

File: system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp (hunk 3 above)
Comment (iwatake2222) [reply system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp]: Fixed in fad266d995c46e12f8ff763e4db67cb19a64298d

File: system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp (hunk 4 above)
Comment (iwatake2222) [reply system/system_monitor/reader/traffic_reader/traffic_reader_service.cpp]: Fixed in fad266d995c46e12f8ff763e4db67cb19a64298d

**Comment by iwatake2222** [comment]
@TetsuKawa 
Thank you for reviewing it. I fixed the code as you suggested.
Could you review it again please?

Also, some workflows are pending and it looks they requires approval from a maintainer. Could you remove/add the tag again?

**Review by TetsuKawa — APPROVED** [review]
LGTM

**Pull request merged**